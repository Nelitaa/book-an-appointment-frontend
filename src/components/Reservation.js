import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import '../styles/reservations.css';
import { fetchDoctors } from '../redux/doctors';

const Reservation = ({ reservation, doctor, onDelete }) => {
  const dispatch = useDispatch();
  // const doctor = doctors.find((doc) => doc.id === doctorId);

  useEffect(() => {
    dispatch(fetchDoctors());
  }, [dispatch]);

  const handleDelete = () => {
    onDelete(reservation.id);
  };

  if (!doctor) {
    return <>Loading</>;
  }

  return (
    <li>
      <h2>{doctor.name}</h2>
      <p className="reservation-date">{reservation.date}</p>
      <p className="reservation-city">{reservation.city}</p>
      <button type="button" onClick={handleDelete}>Cancel Reservation</button>
    </li>
  );
};

Reservation.propTypes = {
  reservation: PropTypes.shape({
    id: PropTypes.number.isRequired,
    user_id: PropTypes.number.isRequired,
    doctor_id: PropTypes.number.isRequired,
    date: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired,
  }).isRequired,
  // doctorId: PropTypes.number.isRequired,
  doctor: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
  onDelete: PropTypes.func.isRequired,
};

export default Reservation;
