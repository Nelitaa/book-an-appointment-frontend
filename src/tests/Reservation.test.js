import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Reservation from '../components/Reservation';

const reservation = {
  id: 1,
  user_id: 1,
  doctor_id: 1,
  date: '2023-04-20',
  city: 'New York',
};

const doctor = {
  id: 1,
  name: 'John Doe',
};

it('Reservation component should render correctly', () => {
  const onDelete = jest.fn();
  const result = render(
    <React.StrictMode>
      <Router>
        <Provider store={store}>
          <Reservation reservation={reservation} doctor={doctor} onDelete={onDelete} />
        </Provider>
      </Router>
    </React.StrictMode>,
  );
  expect(result).toMatchSnapshot();
});
