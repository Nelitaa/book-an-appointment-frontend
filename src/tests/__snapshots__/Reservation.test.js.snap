// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="reservation-card"
      >
        <img
          alt="doctor"
          class="reservation-image"
        />
        <h2
          class="reservation-name"
        >
          Dr.
           
          John Doe
        </h2>
        <img
          alt="dots-bar"
          class="dots-bar"
          src="dots.png"
        />
        <p>
          2023-04-20
        </p>
        <p
          class="reservation-city"
        >
          Based in
           
          New York
        </p>
        <button
          class="reservation-details"
          type="button"
        >
          Cancel
        </button>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="reservation-card"
    >
      <img
        alt="doctor"
        class="reservation-image"
      />
      <h2
        class="reservation-name"
      >
        Dr.
         
        John Doe
      </h2>
      <img
        alt="dots-bar"
        class="dots-bar"
        src="dots.png"
      />
      <p>
        2023-04-20
      </p>
      <p
        class="reservation-city"
      >
        Based in
         
        New York
      </p>
      <button
        class="reservation-details"
        type="button"
      >
        Cancel
      </button>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
